pkg/db-repo/repository.go

```diff
@@ -357,6 +371,120 @@ func ignoreCreatedAtIfNeeded(scope *gorm.Scope) {
 	}
 }
 
+// parses the []uint8 (from mysql) for field x, if there is a field xSet with type []string
+func GormParseMysqlSetType(scope *gorm.Scope) {
+	vo := reflect.ValueOf(scope.Value)
+	if vo.Kind() == reflect.Invalid {
+		return
+	}
+	for vo.Kind() == reflect.Ptr {
+		vo = vo.Elem()
+	}
+	if vo.Kind() == reflect.Slice {
+		for i := 0; i < vo.Len(); i++ {
+			parseMysqlSetType(vo.Index(i))
+		}
+		return
+	}
+	parseMysqlSetType(vo)
+}
+
+func parseMysqlSetType(vo reflect.Value) {
+	if vo.Kind() != reflect.Struct {
+		return
+	}
+	vt := reflect.TypeOf(vo.Interface())
+
+	for i := 0; i < vt.NumField(); i++ {
+		vtf := vt.Field(i)
+		vof := vo.Field(i)
+		fn := vtf.Name
+
+		testType := reflect.TypeOf(make([]string, 0))
+		if !vof.Type().AssignableTo(testType) {
+			continue
```

Comment:
If we have some struct as a member of another struct (foreign key stuff and so on), will we visit it?

```go
type Foo struct {
    ....
    BarId *uint `gorm:...`
    Bar   *Bar
}

type Bar struct {
    Field []uint8
    FieldSet []string
}
```

And now we get query for a `Foo`...

Reply:
i did not check that, however i do expect that tbd as this relation is being pulled by the same mechanisms within gorm, isn't it?

Reply:
If so, why do we need the handling for slices? But sounds reasonable that gorm should handle this... Can we have some integration test for this?